instruqt-tracks/nomad-integration-with-vault/track.yml

```diff
@@ -0,0 +1,490 @@
+        "plugin_name": "postgresql-database-plugin",
+        "allowed_roles": "accessdb",
+        "connection_url": "postgresql://{{username}}:{{password}}@database.service.consul:5432/postgres?sslmode=disable",
+        "username": "demo",
+        "password": "demo"
+      }
+      </pre>
+      <p>The information above allows Vault to connect to our database and create users with specific privileges. We will specify the accessdb role soon. In a production setting, it is recommended to give Vault credentials with enough privileges to generate database credentials dynamically and and manage their lifecycle.</p>
+  - type: text
+    contents: |-
+      <p>Recall from the previous step that we specified accessdb in the allowed_roles key of our connection information.</p>
+      <pre style="overflow-x: auto; white-space: pre-wrap; white-space: -moz-pre-wrap; white-space: -pre-wrap; white-space: -o-pre-wrap; word-wrap: break-word; padding: 15px;">
+      CREATE USER "{{name}}" WITH ENCRYPTED PASSWORD '{{password}}' VALID UNTIL '{{expiration}}';
+      ALTER USER "{{name}}" WITH SUPERUSER;
+      </pre>
+      <p>The SQL above will be used in the creation_statements parameter of our next command to specify the privileges that the dynamic credentials being generated will possess. In our case, the dynamic database user will have broad privileges that include the ability to read from the tables that our application will need to access.</p>
+  - type: text
+    contents: |-
+      <p>Finally, we need to specify a policy named access-tables in our allowed_policies section of the token role. We will create this policy now and give it the capability to read from the database/creds/accessdb endpoint. We will then specify this policy in our Nomad job which will allow it to retrieve credentials for itself to access the database.</p>
+      <pre style="overflow-x: auto; white-space: pre-wrap; white-space: -moz-pre-wrap; white-space: -pre-wrap; white-space: -o-pre-wrap; word-wrap: break-word; padding: 15px;">
+      path "database/creds/accessdb" {
+        capabilities = ["read"]
+      }
+      </pre>
+  tabs:
+  - title: Server
+    type: terminal
+    hostname: hashistack-server
+  - title: Files
+    type: code
+    hostname: hashistack-server
+    path: /root/hashistack/
+  - title: Nomad
+    type: service
+    hostname: hashistack-server
+    port: 4646
+  - title: Consul
+    type: service
+    hostname: hashistack-server
+    port: 8500
+  - title: Vault
+    type: service
+    hostname: hashistack-server
+    port: 8200
+  difficulty: basic
+  timelimit: 300
+- slug: deploy-an-application
+  id: 4ddvvt3xlnzr
+  type: challenge
+  title: Deploy an Application
+  teaser: Deploy an Application with the Appropriate Policy and Templating
+  assignment: |-
+    The Nomad job file has been also been created for you and you can take a look at it by clicking on the <b>"Files"</b> tab and navigating to <b>hashistack > vault > web-app.nomad</b>.</ br></ br>
```

Comment:
I would say more about the details of the template stanza here (or in your notes).  This is at the heart of the Nomad/Vault integration (via Consul Template built into Nomad)

Reply:
I would still reference the actual template portion of the nomad job specification file and explain where it is getting secrets from Vault.